lib/pushmi_pullyu/aip.rb

```diff
@@ -1,63 +1,36 @@
-require 'pushmi_pullyu'
-require 'bagit'
-require 'archive/tar/minitar'
+require 'fileutils'
 
 module PushmiPullyu::AIP
-  # Exceptions
-  class BagInvalid < StandardError; end
-  class NoContentFilename < StandardError; end
-  class FedoraFetchError < StandardError; end
-  class SolrFetchError < StandardError; end
-  class WorkdirInvalid < StandardError; end
   class NoidInvalid < StandardError; end
 
-  def self.create(noid, should_clean_work_directories: true)
-    # Note: returns the filename of the created AIP tarball
-    validate(noid)
-    PushmiPullyu::AIP::Creator
-      .new(noid)
-      .run(should_clean_work_directories: should_clean_work_directories)
+  def self.create(noid)
+    validate_noid(noid)
```

Comment:
Yeah forsure, much agree. Don't like this currently either as you be validating the noid multiple times. Will refactor to your block idea specified in main review comment. Then we don't need to do this.